Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -476,6 +477,42 @@ fileprivate extension Array<AnnotatedTestItem> {
     }
     return result
   }
+
+  func prefixTestsWithModuleName(workspace: Workspace) async -> Self {
+    return await self.asyncMap({
+      // If the module name can't be determined we return the test item without a prefixed id.
+      guard let moduleName = await self.moduleName(from: workspace, for: $0.testItem.location.uri) else {
+        return $0
+      }
+      var newTest = $0.testItem
+      newTest.id = "\(moduleName).\(newTest.id)"
+      newTest.children = await prefixTestsWithModuleName(workspace: workspace, newTest.children)
+      return AnnotatedTestItem(testItem: newTest, isExtension: $0.isExtension)
+    })
+  }
+
+  private func prefixTestsWithModuleName(workspace: Workspace, _ tests: [TestItem]) async -> [TestItem] {
+    return await tests.asyncMap({
+      guard let moduleName = await self.moduleName(from: workspace, for: $0.location.uri) else {
+        return $0
+      }
+
+      var newTest = $0
+      newTest.id = "\(moduleName).\(newTest.id)"
+      newTest.children = await prefixTestsWithModuleName(workspace: workspace, newTest.children)
+      return newTest
+    })
+  }
+
+  private func moduleName(from workspace: Workspace, for uri: DocumentURI) async -> String? {
```

Comment:
Having this method on `Array<AnnotatedTestItem>` seems a little odd to me. If we shared the implementation logic of the two `prefixTestsWithModuleName` methods, I think we could just move this logic into `prefixTestsWithModuleName`.